sdk/src/main/java/io/opentdf/platform/sdk/nanotdf/ECKeyPair.java

```diff
@@ -0,0 +1,232 @@
+package io.opentdf.platform.sdk.nanotdf;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.asn1.x509.SubjectPublicKeyInfo;
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.crypto.digests.SHA256Digest;
+import org.bouncycastle.crypto.generators.HKDFBytesGenerator;
+import org.bouncycastle.crypto.params.HKDFParameters;
+import org.bouncycastle.jce.ECNamedCurveTable;
+import org.bouncycastle.jce.interfaces.ECPrivateKey;
+import org.bouncycastle.jce.interfaces.ECPublicKey;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.jce.spec.ECNamedCurveParameterSpec;
+import org.bouncycastle.math.ec.ECPoint;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.bouncycastle.util.io.pem.*;
+import org.bouncycastle.util.io.pem.PemReader;
+import org.bouncycastle.jce.spec.ECPublicKeySpec;
+
+
+import javax.crypto.KeyAgreement;
+import java.io.*;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.security.spec.*;
+// https://www.bouncycastle.org/latest_releases.html
+
+public class ECKeyPair {
+
+    static {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+    public enum ECAlgorithm {
+        ECDH,
+        ECDSA
+    }
+
+    public enum NanoTDFECCurve {
+        SECP256R1("secp256r1"),
+        PRIME256V1("prime256v1"),
+        SECP384R1("secp384r1"),
+        SECP521R1("secp521r1");
+
+        private String name;
+
+        NanoTDFECCurve(String curveName) {
+            this.name = curveName;
+        }
+
+        @Override
+        public String toString() {
+            return name;
+        }
+    }
+    private KeyPair keyPair;
+    private String curveName;
+
+    public ECKeyPair() throws NoSuchAlgorithmException, NoSuchProviderException, InvalidAlgorithmParameterException {
+        this("secp256r1", ECAlgorithm.ECDH);
+    }
+
+    public ECKeyPair(String curveName, ECAlgorithm algorithm) throws NoSuchAlgorithmException, InvalidAlgorithmParameterException,
+            NoSuchProviderException {
```

Comment:
the signature says it throws those exceptions, but I don't see where they would be coming from.  It *should be* checking parms and throwing those as appropriate though.